### PR TITLE
[MID-17] add migrations docs and update dev-deploy with migrations

### PR DESCRIPTION
**Pull request by alex-popov-tech**
https://trello.com/c/ZRYgL12Z/17-%D0%B7%D0%BC%D1%96%D0%BD%D0%B8-%D0%B2-readme-migration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,40 +2,33 @@ This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next
...
+#### Requirements
 
-Linting is enabled on pre-commit hook, and manually you can run it with `npm run lint` or
-`npm run lint:fix` to automatically apply fixes.
+- Node.js 20+
+- Postgres 16+ running somewhere, and `.env` database variables populated
 
-## Git conventions & guides
+1. `npm install`
+2. `npm run dev`
```
Comment (BohdanLeshchyshyn) [inline README.md]: I guess we need one more step between these two for running migrations.

File: docs/HOWTO.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+- for other database locations - run `npm run migrations`
+
+### How to create a migration
+
+Change database schema and run `npm run migrations`.
+
+Consider following when creating migration:
+
+- using meaningful name
+- separate dropping/creating tables and columns
```
Comment (BohdanLeshchyshyn) [inline docs/HOWTO.md]: It would be better to add more information here
- where to place migrations
- what files it consists of
- how it has to be named - e.g. "<timestamp>_<custom_name>"
- how to generate migration - maybe prisma has some script for creating migration.
- how to roll-back migration in case of any errors
 
 I think it has to be required also with migration to create a file for its roll-back.

File: .github/workflows/dev-deploy.yaml [hunk 3]
```diff
@@ -17,5 +19,7 @@ jobs:
         run: vercel pull --yes --environment=production --token=${{ secrets.DEV_VERCEL_TOKEN }}
       - name: Build Project Artifacts
         run: vercel build --prod --token=${{ secrets.DEV_VERCEL_TOKEN }}
+      - name: Run Migrations
+        run: npm run migrations
```
Comment (BohdanLeshchyshyn) [inline .github/workflows/dev-deploy.yaml:23]: What do you think, is it possible to also add to GitHub actions manual jobs for running migration up and down in case of some issues?

**Review by BohdanLeshchyshyn — COMMENTED** [review]
Good job!

File: README.md (hunk 1 above)
Comment (alex-popov-tech) [reply README.md]: added

File: docs/HOWTO.md (hunk 2 above)
Comment (alex-popov-tech) [reply docs/HOWTO.md]: `where to place migrations`
they are generated in default locations, there is no ability to choose target folder

`what files it consists of`
there is single `sql` file, these generally should not be edited manually, what could be documented about it? 🤔 

`how it has to be named - e.g. "_<custom_name>"`
i think prisma automatically applies its formating rules for migration written in plain english

`how to generate migration`
there is one command for generationg and running migrations, i will update doc to explicitly note that point

`how to roll-back migration in case of any errors`
there is no such ability - only reset completely. i will add a point about that too

File: .github/workflows/dev-deploy.yaml (hunk 3 above)
Comment (alex-popov-tech) [reply .github/workflows/dev-deploy.yaml:23]: that depends on actual use cases, i think there is ability to have manually triggered jobs

but that is waaaay out of initial scope :)
